### PR TITLE
Remove CFR_ACC_VALUE_TYPE and CFR_ACC_PRIMITIVE_VALUE_TYPE

### PR DESCRIPTION
**Pull request by theresa-m**

### CODE REVIEW

**Comment by hangshao0** [comment]
The line ending check failed.

File: runtime/cfdumper/main.c [hunk 1]
```diff
@@ -1113,9 +1114,9 @@ static void printMethod(J9CfrClassFile* classfile, J9CfrMethod* method)
 	if(method->accessFlags & CFR_ACC_STRICT) j9tty_printf( PORTLIB, "strict ");
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
 	/* ImplicitCreation is triggered by an implicit constructor in a value class */
-	if (classfile->accessFlags & CFR_ACC_VALUE_TYPE) {
+	if (isClassValueType(classfile)) {
 		J9CfrConstantPoolInfo name = classfile->constantPool[method->nameIndex];
-		if (J9UTF8_LITERAL_EQUALS(name.bytes, name.slot1, "<vnew>")) {
+		if (J9UTF8_LITERAL_EQUALS(name.bytes, name.slot1, "<new>")) {
```
Comment (hangshao0) [inline runtime/cfdumper/main.c:1119]: Isn't `<vnew>` replaced by `<init>` ?

File: runtime/nls/cfre/cfrerr_ca.nls [hunk 2]
```diff
@@ -1434,13 +1434,6 @@ J9NLS_CFR_ERR_OUTER_CLASS_BAD_ARRAY_CLASS.system_action=The JVM will throw a ver
 J9NLS_CFR_ERR_OUTER_CLASS_BAD_ARRAY_CLASS.user_response=Contact the provider of the classfile for a corrected version.
 # END NON-TRANSLATABLE
 
-J9NLS_CFR_ERR_VALUE_FLAG_MISSING_ON_PRIMITIVE_CLASS=Falta el senyalador d'acc\u00e9s ACC_VALUE a la classe primitiva.
```
Comment (hangshao0) [inline runtime/nls/cfre/cfrerr_ca.nls:1437]: Please revert the changes in this file.

File: runtime/oti/cfr.h [hunk 3]
```diff
@@ -923,22 +921,15 @@ typedef struct J9CfrClassFile {
 #define CFR_MINOR_VERSION  3
 #define CFR_PUBLIC_PRIVATE_PROTECTED_MASK	(CFR_ACC_PUBLIC | CFR_ACC_PRIVATE | CFR_ACC_PROTECTED)
 
+#define VALUE_TYPES_MAJOR_VERSION 67
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_IDENTITY | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM | CFR_ACC_VALUE_TYPE | CFR_ACC_PRIMITIVE_VALUE_TYPE)
-#define J9_IS_CLASSFILE_VALUETYPE(classfile)     J9_ARE_ALL_BITS_SET((classfile)->accessFlags, CFR_ACC_VALUE_TYPE)
-#define CFR_INTERFACE_CLASS_ACCESS_MASK			(CFR_ACC_PUBLIC | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_VALUE_TYPE | CFR_ACC_IDENTITY)
+#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_IDENTITY | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM)
+#define CFR_INTERFACE_CLASS_ACCESS_MASK			(CFR_ACC_PUBLIC | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_IDENTITY)
```
Comment (hangshao0) [inline runtime/oti/cfr.h:927]: `CFR_ACC_IDENTITY` should be removed from `CFR_INTERFACE_CLASS_ACCESS_MASK`
The new [spec](https://cr.openjdk.org/~dlsmith/jep401/jep401-20240603/specs/value-objects-jvms.html#jvms-4.1) says:

`If the ACC_INTERFACE flag is set......ACC_IDENTITY, ACC_ENUM, and ACC_MODULE flags set must not be set.`

File: runtime/cfdumper/main.c (hunk 1 above)
Comment (theresa-m) [reply runtime/cfdumper/main.c:1119]: yes, you're right.

File: runtime/bcutil/cfreader.c [hunk 4]
```diff
@@ -65,6 +65,9 @@ static void sortMethodIndex(J9CfrConstantPoolInfo* constantPool, J9CfrMethod *li
 static IDATA compareMethodIDs(J9CfrConstantPoolInfo* constantPool, J9CfrMethod *a, J9CfrMethod *b);
 static U_8* getUTF8Data(J9CfrConstantPoolInfo* constantPool, U_16 cpIndex);
 static U_16 getUTF8Length(J9CfrConstantPoolInfo* constantPool, U_16 cpIndex);
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+static BOOLEAN isClassValueType(J9CfrClassFile * classfile);
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/bcutil/cfreader.c]: I have moved it to a macro.

File: runtime/oti/cfr.h [hunk 5]
```diff
@@ -923,22 +921,12 @@ typedef struct J9CfrClassFile {
...
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-#define J9_IS_CLASSFILE_PRIMITIVE_VALUETYPE(classfile)     J9_ARE_ALL_BITS_SET((classfile)->accessFlags, CFR_ACC_PRIMITIVE_VALUE_TYPE)
-#else /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
-#define J9_IS_CLASSFILE_PRIMITIVE_VALUETYPE(classfile) FALSE
-#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
+#define VALUE_TYPES_MAJOR_VERSION 67
+#define PREVIEW_MINOR_VERSION 65535
+#define J9_IS_CLASSFILE_VALUETYPE(classfile) ((classfile->majorVersion >= VALUE_TYPES_MAJOR_VERSION) && (PREVIEW_MINOR_VERSION == classfile->minorVersion) && J9_ARE_NO_BITS_SET(classfile->accessFlags, CFR_ACC_IDENTITY))
 
+#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_IDENTITY | CFR_ACC_SUPER | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM)
```
Comment (hangshao0) [inline runtime/oti/cfr.h:928]: Even `CFR_ACC_IDENTITY` and `CFR_ACC_SUPER` share the same bit, not everyone reading this code is aware of that. So it is better to split this macro for VT and non-VT builds, so that they won't be confused by these 2 flags. i.e.

```
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
//define CFR_CLASS_ACCESS_MASK with CFR_ACC_IDENTITY
#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
//define CFR_CLASS_ACCESS_MASK with CFR_ACC_SUPER
#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```

File: runtime/util/rcdump.c [hunk 6]
```diff
@@ -96,6 +96,10 @@ static I_32 dumpCallSiteData ( J9PortLibrary *portLib, J9ROMClass *romClass, U_3
 static I_32 dumpStaticSplitSideTable( J9PortLibrary *portLib, J9ROMClass *romClass, U_32 flags);
 static I_32 dumpSpecialSplitSideTable( J9PortLibrary *portLib, J9ROMClass *romClass, U_32 flags);
 static void printMethodExtendedModifiers(J9PortLibrary *portLib, U_32 modifiers);
+
+BOOLEAN isRomClassValueType(J9ROMClass *romClass) {
```
Comment (hangshao0) [inline runtime/util/rcdump.c:100]: I believe macro `J9ROMCLASS_IS_VALUE()` needs to be updated to include the version checks. You can use `J9ROMCLASS_IS_VALUE()` in this file instead of adding this new function.

File: runtime/oti/cfr.h (hunk 5 above)
Comment (theresa-m) [reply runtime/oti/cfr.h:928]: Okay. What about a comment to that effect instead?

File: runtime/oti/cfr.h [hunk 7]
```diff
@@ -923,22 +921,14 @@ typedef struct J9CfrClassFile {
 #define CFR_MINOR_VERSION  3
 #define CFR_PUBLIC_PRIVATE_PROTECTED_MASK	(CFR_ACC_PUBLIC | CFR_ACC_PRIVATE | CFR_ACC_PROTECTED)
 
+#define J9_IS_CLASSFILE_VALUETYPE(classfile) ((classfile->majorVersion >= VALUE_TYPES_MAJOR_VERSION) && (PREVIEW_MINOR_VERSION == classfile->minorVersion) && J9_ARE_NO_BITS_SET(classfile->accessFlags, CFR_ACC_IDENTITY))
```
Comment (hangshao0) [inline runtime/oti/cfr.h:924]: Please change `classfile->majorVersion`, `classfile->minorVersion` and `classfile->accessFlags` to
` (classfile)->majorVersion`, `(classfile)->minorVersion` and `(classfile)->accessFlags`

File: runtime/util/rcdump.c (hunk 6 above)
Comment (hangshao0) [reply runtime/util/rcdump.c]: I don't see the major and minor version checks added to `J9ROMCLASS_IS_VALUE()`.

**Comment by hangshao0** [comment]
Jenkins compile zlinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins compile amac jdknext

**Review by hangshao0 — APPROVED** [review]

File: runtime/bcutil/cfreader.c [hunk 8]
```diff
@@ -3140,19 +3140,10 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 	/* Currently value type is built on JDK22, so compare with JDK22 for now. */
```
Comment (keithc-ca) [inline runtime/bcutil/cfreader.c:3142]: This comment should have been updated with the change on line 3143.

Comment (theresa-m) [reply runtime/bcutil/cfreader.c:3142]: Thanks Keith. I have another pull request in the works to close out https://github.com/eclipse-openj9/openj9/issues/18829 I will update the comment there.
